bin/pygrb/pycbc_pygrb_page_tables

```diff
@@ -266,84 +287,90 @@ for output_file in output_files:
-logging.info("Loading timeslides.")
-slide_dict = ppu.load_time_slides(offsource_file)
-logging.info("Loading segments.")
-segment_dict = ppu.load_segment_dict(offsource_file)
+# Extract IFOs
+ifos = ppu.extract_ifos(offsource_file)
 
-# Calculate chirp masses of templates
-logging.info('Loading triggers template masses')
-bank_data = HFile(opts.bank_file, 'r')
-mchirps = mchirp_from_mass1_mass2(
-        bank_data['mass1'][...],
-        bank_data['mass2'][...]
-    )
+# Generate time-slides dictionary
+slide_dict = ppu.load_time_slides(offsource_file)
 
-# Construct trials
-logging.info("Constructing trials.")
-trial_dict = ppu.construct_trials(opts.seg_files, segment_dict,
-                                  ifos, slide_dict, vetoes)
-total_trials = sum([len(trial_dict[slide_id]) for slide_id in slide_dict])
-logging.info("%d trials generated.", total_trials)
-
-# Extract basic trigger properties and store as dictionaries
-trig_time, trig_snr, trig_bestnr = \
-    ppu.extract_basic_trig_properties(trial_dict, trig_data, slide_dict,
-                                      segment_dict, opts)
-# Calculate SNR and BestNR values and maxima
-time_veto_max_snr = {}
-time_veto_max_bestnr = {}
-for slide_id in slide_dict:
-    num_slide_segs = len(trial_dict[slide_id])
-    time_veto_max_snr[slide_id] = np.zeros(num_slide_segs)
-    time_veto_max_bestnr[slide_id] = np.zeros(num_slide_segs)
+# Generate segments dictionary
+segment_dict = ppu.load_segment_dict(offsource_file)
 
+# Construct trials removing vetoed times
+trial_dict, total_trials = ppu.construct_trials(opts.seg_files, segment_dict,
+                                                ifos, slide_dict,
+                                                opts.veto_file)
+
+# Load triggers (apply reweighted SNR cut, not vetoes)
+trig_data = ppu.load_data(offsource_file, ifos, data_tag='offsource',
+                          rw_snr_threshold=opts.newsnr_threshold,
+                          slide_id=opts.slide_id)
+
+# Extract needed trigger properties and store them as dictionaries
+# Based on trial_dict: if vetoes were applied, trig_* are the veto survivors
+# _av stands for after vetoes
+keys = ['network/end_time_gc', 'network/coherent_snr', 'network/reweighted_snr']
+trig_data_av = ppu.extract_trig_properties(
+    trial_dict,
+    trig_data,
+    slide_dict,
+    segment_dict,
+    keys
+)
+
+# Max SNR and BestNR values in each trial: these are stored in dictionaries
+# keyed by slide_id, as arrays indexed by trial number
+background_snr = {k: np.zeros(len(v)) for k,v in trial_dict.items()}
+background = {k: np.zeros(len(v)) for k,v in trial_dict.items()}
 for slide_id in slide_dict:
     for j, trial in enumerate(trial_dict[slide_id]):
-        trial_cut = (trial[0] <= trig_time[slide_id])\
-                          & (trig_time[slide_id] < trial[1])
+        # True whenever the trigger is in the trial
+        trial_cut = (trial[0] <= trig_data_av[keys[0]][slide_id][:])\
```

Comment:
Similar to a comment I made elsewhere earlier: define a local helper variable that holds `trig_data_av[keys[0]][slide_id][:]` just before this loop.